### DISCUSSION TITLE
BasicTree组件，怎么拿到halfCheckedKey

### DISCUSSION THREAD

**Discussion by uncarbon97** [vbenjs/vue-vben-admin#2748] [Q&A]
双向绑定 checkedKeys 只能拿到完全勾选的
半勾选（一般是父级节点）应该怎么拿到呢？

另外问一下 ref 取DOM的正确写法应该是什么……网络上好多教程看了 结果都是null

**Comment by ghost** [comment]
> 双向绑定 checkedKeys 只能拿到完全勾选的，半勾选（一般是父级节点）应该怎么拿到呢？

这是antdv的文档，可以参考一下

<img width="916" alt="image" src="https://user-images.githubusercontent.com/106857035/235823228-3c3a9575-1701-49d6-9079-607b4e1a8c5b.png">

> 另外问一下 ref 取DOM的正确写法应该是什么……网络上好多教程看了 结果都是null

下面是vue3中的写法，可以参考一下
```vue
<template>
  <div ref="myRef">Hello World!</div>
</template>

<script>
import { ref, onMounted } from 'vue';

export default {
  setup() {
    const myRef = ref(null);

    onMounted(() => {
      console.log(myRef.value);
    });

    return {
      myRef
    };
  }
};
</script>
```
